src/metrics/loc.rs

```diff
@@ -880,6 +889,60 @@ mod tests {
         );
     }
 
+    #[test]
+    fn cpp_lloc() {
+        check_metrics!(
+            "nsTArray<xpcGCCallback> callbacks(extraGCCallbacks.Clone());
+             for (uint32_t i = 0; i < callbacks.Length(); ++i) {
+                 callbacks[i](status);
+             }",
+            "foo.cpp",
+            CppParser,
+            loc,
+            [(lloc, 3, usize)] // nsTArray, for, callbacks
+        );
+    }
+
+    #[test]
+    fn cpp_return_lloc() {
+        check_metrics!(
+            "uint8_t* pixel_data = frame.GetFrameDataAtPos(DesktopVector(x, y));
+             return RgbaColor(pixel_data) == blank_pixel_;",
+            "foo.cpp",
+            CppParser,
+            loc,
+            [(lloc, 2, usize)] // pixel_data, return
+        );
+    }
+
+    #[test]
+    fn cpp_for_lloc() {
+        check_metrics!(
+            "for (; start != end; ++start) {
+                 const unsigned char idx = *start;
+                 if (idx > 127 || !kValidTokenMap[idx]) return false;
+             }",
+            "foo.cpp",
+            CppParser,
+            loc,
+            [(lloc, 4, usize)] // for, idx, if, return
```

Comment:
Shouldn't these be actually 3? The return is on the same line as the if

Reply:
`lloc` counts by statements, so it's correct. An `if` is a statement and a `return` another one